### PR TITLE
UTs: Bump Roslyn to 4.9.2

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**

### CODE REVIEW

**Review by antonioaversa — APPROVED** [review]
LGTM
All Razor-related changes seem to be OK: at the beginning I though that during Blazor rules implementation, we had put in place code compensating for the off-by-1 locations, but it wasn't actually the case. So all good.
I left a non-blocking remark and a couple of questions/heads-up.

File: analyzers/tests/SonarAnalyzer.Test/Rules/Utilities/SymbolReferenceAnalyzerTest.cs [hunk 1]
```diff
@@ -242,8 +242,8 @@ public void Verify_UnchangedFiles(string unchangedFileName, bool expectedProtobu
                         // Net8 SDK: Declaration (1,0) - (1,17) Reference (1,6) - (1,23) <- Overlapping
                         // Declaration of TSomeVeryLongName is placed starting at index 0 (ignoring "@typeparam ")
                         // Reference "where TSomeVeryLongName" is placed starting at index 6 (ignoring "@typeparam TSomeVeryLongName ")
-                        // In Net7 SDK, no references are produced
-                        orderedSymbols[3].Reference.Should().BeEmpty();
+                        VerifyReferences(orderedSymbols[3].Reference, 1, 1, 1);
```
[the remark this answers is not in the document]
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/Rules/Utilities/SymbolReferenceAnalyzerTest.cs:245]: See https://github.com/SonarSource/sonar-dotnet/issues/8417#issuecomment-1987850097

> Did you open a ticket on roslyn side?

I can not remember as there was some email contact at the time between us and Microsoft. We had a list of bugs but I do not know whether this one was included. I think @andrei-epure-sonarsource was in charge then. Maybe he knows where that list is.

[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/Rules/Utilities/SymbolReferenceAnalyzerTest.cs:245]: This is what we have:
- https://github.com/dotnet/razor/issues/9050
- https://github.com/dotnet/razor/issues/9051
- https://github.com/dotnet/razor/issues/9108
- https://github.com/dotnet/roslyn/issues/69248
- https://github.com/dotnet/razor/issues/9308
- https://github.com/dotnet/roslyn/issues/71449

Please open an issue if needed.

[the remark this answers is not in the document]
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/Rules/Utilities/SymbolReferenceAnalyzerTest.cs:245]: It's not on the list. I created https://github.com/dotnet/razor/issues/10087

**Comment by pavel-mikula-sonarsource** [comment]
Converted to draft to prevent merging before release.

File: analyzers/tests/SonarAnalyzer.TestFramework/Verification/Verifier.cs [hunk 2]
```diff
@@ -362,7 +362,8 @@ private void ValidateCodeFix()
     }
 
     private static bool IsRazorOrCshtml(string path) =>
-        Path.GetExtension(path) is var extension
+        path is not null
+        && Path.GetExtension(path) is var extension
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.TestFramework/Verification/Verifier.cs:366]: Good point with the `null` transitivity.

It's not a Roslyn change, it's the `AddSnippet` oversight from the time the Razor assertion was originally implemented. Snippet has a `null` filename by default. And this method did not expect that.

File: analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Constants.cs [hunk 3]
```diff
@@ -67,9 +67,9 @@ public void PreProcess_EqualityCheck_BooleanConstraint(string constant, bool val
             $"LocalReference: a = 1 == {constant} (Implicit)",
             "Literal: 1",
             $"Literal: {constant}",
-            $"BinaryOperator: 1 == {constant}",
+            $"Binary: 1 == {constant}",
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Constants.cs:70]: I don't know. This part hasn't changed for 6 years now
https://github.com/dotnet/roslyn/blob/ff27b0876a5b38d1bcc503ae763f81e5318ff18a/src/Compilers/Core/Portable/Generated/OperationKind.Generated.cs#L83-L87

**Pull request merged**